api/src/test/java/org/openmrs/DrugOrderTest.java

```diff
@@ -314,4 +314,95 @@ public void setAutoExpireDateBasedOnDuration_shouldNotCalculateIfAutoExpireDateA
 		
 		assertEquals(expectedAutoExpireDate, drugOrder.getAutoExpireDate());
 	}
+
+	/**
+	 * @verifies return true if the drugs match and drug non coded match
+	 * @see DrugOrder#hasSameOrderableAs(Order)
+	 */
+	@Test
+	public void hasSameOrderableAs_shouldReturnTrueIfTheConceptsMatchAndHaveSameDrugNonCoded() throws Exception {
+		DrugOrder order = new DrugOrder();
+		Concept concept = new Concept();
+		Drug drug1 = new Drug();
+		drug1.setConcept(concept);
+		order.setDrug(drug1);
```

Comment:
I thought drug should be null for a nonCoded order, but i see you are setting it here